includes/class-newspack-popups-model.php

```diff
@@ -12,6 +12,87 @@
  */
 final class Newspack_Popups_Model {
 
+	/**
+	 * Retrieve all Popus (first 100).
+	 *
+	 * @return array Array of Popup objects.
+	 */
+	public static function retrieve_popups() {
+		$args = [
+			'post_type'      => Newspack_Popups::NEWSPACK_PLUGINS_CPT,
+			'post_status'    => 'publish',
+			'posts_per_page' => 100,
+		];
+
+		$popups = self::retrieve_popup_with_query( new WP_Query( $args ), true );
+
+		// Hacky check if this is a single popup object as opposed to an array of Pop-ups. TODO: replace.
+		if ( isset( $popups['id'] ) ) {
```

Comment:
```suggestion	
		if ( $popups && ! is_array( $popups ) ) {
```

Here's the answer to your TODO :)

Reply:
> Here's the answer to your TODO :)

I believe `is_array()` will return true if it's an associative array, so that won't catch this case.

Reply:
The new logic in https://github.com/Automattic/newspack-popups/pull/34/commits/a7ca443afb45b858255fdfccb81b4348e60ab376 makes this bit unnecessary anyway...